packages/teams-js/src/public/appNotification.ts

```diff
@@ -0,0 +1,187 @@
+   *
+   * @param inputString and maximumLength
+   * @returns True if string length is within specified limit
+   */
+  function isValidLength(inputString: string, maxLength: number): boolean {
+    return inputString.length <= maxLength;
+  }
+
+  /**
+   * Validates that all the required appNotification display parameters are either supplied or satisfy the required checks
+   * @param notificationDisplayParam
+   * @throws Error if any of the required parameters aren't supplied
+   * @throws Error if content or title length is beyond specific limit
+   * @throws Error if invalid url is passed
+   * @returns void
+   */
+  function validateNotificationDisplayParams(notificationDisplayParam: NotificationDisplayParam): void {
+    const maxTitleLength = 75;
+    const maxContentLength = 1500;
+
+    if (!notificationDisplayParam.title) {
+      throw new Error('Must supply notification title');
+    }
+
+    if (!isValidLength(notificationDisplayParam.title, maxTitleLength)) {
+      throw new Error(
+        `Invalid notification title length: Maximum title length ${maxTitleLength}, title length supplied ${notificationDisplayParam.title.length}`,
+      );
+    }
+
+    if (!notificationDisplayParam.content) {
+      throw new Error('Must supply notification content');
+    }
+    if (!isValidLength(notificationDisplayParam.content, maxContentLength)) {
+      throw new Error(
+        `Invalid notification content length: Maximum content length ${maxContentLength}, content length supplied ${notificationDisplayParam.content.length}`,
+      );
+    }
+
+    if (!notificationDisplayParam.notificationActionUrl) {
+      throw new Error('Must supply notification url');
+    }
+    if (!isValidUrl(notificationDisplayParam.notificationActionUrl)) {
+      throw new Error('Invalid notificationAction url');
+    }
+
+    if (notificationDisplayParam.icon !== undefined && !isValidUrl(notificationDisplayParam.icon)) {
+      throw new Error('Invalid icon url');
+    }
+
+    if (!notificationDisplayParam.displayDurationInSeconds) {
+      throw new Error('Must supply display duration in seconds');
+    }
+
+    if (notificationDisplayParam.displayDurationInSeconds <= 0) {
+      throw new Error('Notification display time must be greater than zero');
+    }
+  }
+
+  /**
+   * Displays appNotification after making a validiity check on all of the required parameters, by calling the validateNotificationDisplayParams helper function
+   * An interface object containing all the required parameters to be displayed on the notification would be passed in here
+   * The notificationDisplayParam would be serialized before passing across to the message handler to ensure all objects passed contain simple parameters that would properly pass across the Iframe
+   * @param notificationdisplayparam - Interface object with all the parameters required to display an appNotificiation
+   * @returns a promise resolution upon conclusion
+   * @throws
```

Comment:
>@throws [](http://example.com/codeflow?start=5&length=7)

You'll also need to actually document _what_ this throws, not just that it throws. You can look at other functions in this library as an example, e.g.,

```
     * @throws Error if {@linkcode app.initialize} has not successfully completed
```